### PR TITLE
split tests for CI build into TOPP vs. class-tests

### PR DESCRIPTION
**Pull request by cbielow**

### CODE REVIEW

**Comment by hroest** [comment]
this seems like a good idea, the important part is to either only build the TOPP tools and then test those and in the other case *not* build the TOPP tools and only build the class tests so probably we want to change the `ctest_build` command to only build the corresponding target, correct?

Did I miss this change or was this not yet done, I somehow did not see it in this commit series?

**Comment by cbielow** [comment]
this was not done (yet).
In the meanwhile I switched back to build and test all, but with -O0 for 
gcc (which is the compiler causing trouble). It did indeed shrink the 
time below 50min, but only so slightly that it will not be a permanent 
solution.

So I think we need the test split -- will try this now...

Am 7/2/2015 um 2:06 AM schrieb Hannes Roest:

**Comment by cbielow** [comment]
Ok, we're down to 42 (TOPP only) and 47 minutes (class tests) now.
Not a lot of head space left, but I think worth merging until we have to split the class tests again.

File: tools/lnx-cibuild.sh [hunk 1]
```diff
@@ -31,12 +31,16 @@ if [ "${ENABLE_STYLE_TESTING}" = "On" ]; then
   _build_name=${_build_name}"-coding-style"
 fi
 
-# add unity to build name if requested
-if [ "${ENABLE_UNITYBUILD}" = "On" ]; then
-  _build_name=${_build_name}"-unity-build"
+# add class-testing to build name if requested
+if [ "${ENABLE_CLASS_TESTING}" = "On" ]; then
+  _build_name=${_build_name}"-testClass"
```
Comment (aiche) [inline tools/lnx-cibuild.sh]: just being picky, why not follow the style of the other build names and make it `test-class`? rest looks good. +1 for merge

**Comment by cbielow** [comment]
This should be ready to merge

**Comment by hroest** [comment]
@cbielow there is a major issue here with this commit, travis does not work any more for clang, see #1543

**Pull request merged**